Library/Homebrew/software_spec.rb

```diff
@@ -281,6 +281,10 @@ def bintray
       ERB::Util.url_encode("#{name}-#{version}#{extname}")
     end
 
+    def github_packages
+      "#{name}--#{version}#{extname}"
```

Comment:
This uses two dashes, while the `bintray` name format only uses one dash. We have been following the `bintray` naming pattern in the `osrf/simulation` tap and upload our bottles with one dash to s3 (all bottles viewable at [gazebosim.org/distributions/bottles-simulation](http://gazebosim.org/distributions/bottles-simulation/).

Since this PR was merged, our bottles are being uploaded with two dashes, since we currently upload them using the `bottle.tags.filename` field in the json file ([logic in upload script here](https://github.com/ignition-tooling/release-tools/blob/master/jenkins-scripts/lib/repository_uploader.bash#L161-L163)). 

I saw https://github.com/Homebrew/brew/pull/11097 was made to fix bintray uploads via the `upload_bottles` method in `bintray.rb`, but the `bottle.tags.filename` json field still has two dashes.

So how many dashes are expected in bottle filenames?

Reply:
> So how many dashes are expected in bottle filenames?

This doesn't answer your question (you raise a good quesetion about handling backwards compatbility), but I can at least give some background:

The single dash has mostly been for backwards compatbility - it only applies to the remote filename. When dowloaded, it is actually stored with two dashes. There's no backwards compatibility concern with GitHub Packages, so we do not support single dash on that platform.

Reply:
https://github.com/Homebrew/brew/pull/11106

Reply:
> Related: #11106

thanks for pointing that out; that resolves the issue for me